From 5e314383aa284d531788d4ea1395ef3722b08587 Mon Sep 17 00:00:00 2001
From: Michael S. Tsirkin <mst@redhat.com>
Date: Wed, 21 Jul 2010 14:10:52 -0300
Subject: [PATCH 20/24] Revert "virtio: Enable the PUBLISH_USED feature by default for all devices"

RH-Author: Michael S. Tsirkin <mst@redhat.com>
Message-id: <15d570fe46c8cdb0e080beabba98df7bea2c28bd.1279721307.git.mst@redhat.com>
Patchwork-id: 10895
O-Subject: [PATCH RHEL6.0 1/3] Revert "virtio: Enable the PUBLISH_USED feature
	by default for all devices"
Bugzilla: 616501
RH-Acked-by: Juan Quintela <quintela@redhat.com>
RH-Acked-by: Amit Shah <amit.shah@redhat.com>
RH-Acked-by: Alex Williamson <alex.williamson@redhat.com>

This reverts commit 34bb77bdedba99cf3384d16bacdccf8051aff683.

Signed-off-by: Michael S. Tsirkin <mst@redhat.com>

Part of a series reverting PUBLISH_USED.

PUBLISH_USED was reverted upstream.
Discussions on layout and ABI to use are still ongoing,
and there is high risk layout won't match what we put in RHEL,
so we won't be able to support this going forward.

https://bugzilla.redhat.com/show_bug.cgi?id=616501
https://brewweb.devel.redhat.com/taskinfo?taskID=2614379

Tested: on x86_64

---
 hw/virtio-net.h |    2 ++
 hw/virtio.h     |    4 +---
 2 files changed, 3 insertions(+), 3 deletions(-)

Signed-off-by: Eduardo Habkost <ehabkost@redhat.com>
---
 hw/virtio-net.h |    2 ++
 hw/virtio.h     |    4 +---
 2 files changed, 3 insertions(+), 3 deletions(-)

diff --git a/hw/virtio-net.h b/hw/virtio-net.h
index e55119b..27c63ee 100644
--- a/hw/virtio-net.h
+++ b/hw/virtio-net.h
@@ -155,6 +155,8 @@ struct virtio_net_ctrl_mac {
 
 #define DEFINE_VIRTIO_NET_FEATURES(_state, _field) \
         DEFINE_VIRTIO_COMMON_FEATURES(_state, _field), \
+        DEFINE_PROP_BIT("publish_used", _state, _field, \
+                        VIRTIO_RING_F_PUBLISH_USED, true), \
         DEFINE_PROP_BIT("csum", _state, _field, VIRTIO_NET_F_CSUM, true), \
         DEFINE_PROP_BIT("guest_csum", _state, _field, VIRTIO_NET_F_GUEST_CSUM, true), \
         DEFINE_PROP_BIT("gso", _state, _field, VIRTIO_NET_F_GSO, true), \
diff --git a/hw/virtio.h b/hw/virtio.h
index bcbe94a..8bd75f9 100644
--- a/hw/virtio.h
+++ b/hw/virtio.h
@@ -192,9 +192,7 @@ void virtio_net_exit(VirtIODevice *vdev);
 
 #define DEFINE_VIRTIO_COMMON_FEATURES(_state, _field) \
 	DEFINE_PROP_BIT("indirect_desc", _state, _field, \
-			VIRTIO_RING_F_INDIRECT_DESC, true), \
-        DEFINE_PROP_BIT("publish_used", _state, _field, \
-                        VIRTIO_RING_F_PUBLISH_USED, true)
+			VIRTIO_RING_F_INDIRECT_DESC, true)
 
 target_phys_addr_t virtio_queue_get_desc_addr(VirtIODevice *vdev, int n);
 target_phys_addr_t virtio_queue_get_avail_addr(VirtIODevice *vdev, int n);
-- 
1.7.0.3

